### PR TITLE
[improve][cli] Supports creating tokens with additional headers

### PR DESCRIPTION
**Pull request by coderzc**
### Motivation

Currently, the tool that generates the token cannot attach headers. We should support it. For example, it makes sense to attach a [kid](https://www.rfc-editor.org/rfc/rfc7515#section-4.1.4) field for JWKS.

### Modifications

Supports creating tokens with additional headers

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [x] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtils.java [hunk 1]
```diff
@@ -140,6 +141,11 @@ public static class CommandCreateToken implements Callable<Integer> {
                 description = "Pass the private key for signing the token. This can either be: data:, file:, etc..")
         private String privateKey;
 
+        @Option(names = {"-h",
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtils.java:144]: Please remove "-h", which conflicts with the help command.

Comment (coderzc) [reply pulsar-broker/src/main/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtils.java]: '-h' -> '-hs'

**Review by nodece — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Comment by Technoboy-** [comment]
@lhotari @heesung-sn Could you help take a look ?

File: pulsar-broker/src/test/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtilsTest.java [hunk 2]
```diff
@@ -31,6 +36,37 @@
...
+        try {
+            ByteArrayOutputStream baoStream = new ByteArrayOutputStream();
+            System.setOut(new PrintStream(baoStream));
+
+            new TokensCliUtils().execute(new String[]{"create-secret-key", "--base64"});
+            String secretKey = baoStream.toString();
+
+            baoStream.reset();
+
+            new TokensCliUtils().execute(new String[]{"create", "--secret-key", "data:;base64," + secretKey, "--subject", "test", "--headers", "kid=test"});
```
Comment (heesung-sn) [inline pulsar-broker/src/test/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtilsTest.java:51]: nit: can we add a case with an invalid jwt header ?

https://self-issued.info/docs/draft-jones-json-web-token-01.html#ReservedHeaderParameterName

**Review by heesung-sn — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtilsTest.java (hunk 2 above)
Comment (coderzc) [reply pulsar-broker/src/test/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtilsTest.java]: done

**Pull request merged**